managedtenants/utils/ocm.py

```diff
@@ -265,8 +269,43 @@ def _addon_from_imageset(self, imageset, metadata):
             "channel": metadata.get("defaultChannel"),
         }
 
+        # Set attributes from metdata file if present.
+        # They will get overwritten with the values from the imageset if they're
+        # present in the imageset as well.
+        if metadata.get("pullSecretName"):
+            addon[self.IMAGESET_KEYS["pullSecretName"]] = metadata.get(
+                "pullSecretName"
+            )
+
+        if metadata.get("additionalCatalogSources"):
+            addon[
+                self.IMAGESET_KEYS["additionalCatalogSources"]
+            ] = self._make_additional_ctlg_src_list(
+                metadata.get("additionalCatalogSources")
+            )
+
+        if metadata.get("subscriptionConfig"):
+            addon[self.IMAGESET_KEYS["subscriptionConfig"]] = {}
+            addon[self.IMAGESET_KEYS["subscriptionConfig"]][
+                "add_on_environment_variables"
+            ] = self._make_env_vars_list(
+                metadata.get("subscriptionConfig").get("env")
+            )
+
         for key, val in imageset.items():
             if key in self.IMAGESET_KEYS:
+                if key == "additionalCatalogSources":
```

Comment:
This is a very small thing but it might be a little bit easier to read if we assigned `self.IMAGESET_KEYS[key]` to a variable, ex. `mapped_key = self.IMAGESET_KEYS[key]`